ipc/nodepart/messenger.go

```diff
@@ -0,0 +1,53 @@
+package nodepart
+
+import (
+	"os"
+	"time"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/common"
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/logger"
+)
+
+// NodeMessenger is the messenger on Node's part of the pipe
+type NodeMessenger struct {
+	common.Messenger
+}
+
+// NewNodeMessenger creates a new messenger
+func NewNodeMessenger(logger logger.Logger, reader *os.File, writer *os.File) *NodeMessenger {
+	return &NodeMessenger{
+		Messenger: *common.NewMessenger("NODE", logger, reader, writer),
+	}
+}
+
+// SendContractRequest sends a request to Arwen
+func (messenger *NodeMessenger) SendContractRequest(request common.MessageHandler) error {
+	err := messenger.Send(request)
+	if err != nil {
+		return common.ErrCannotSendContractRequest
+	}
+
+	return nil
+}
+
+// SendHookCallResponse sends a hook response to Arwen
+func (messenger *NodeMessenger) SendHookCallResponse(response common.MessageHandler) error {
+	err := messenger.Send(response)
+	if err != nil {
+		return common.ErrCannotSendHookCallResponse
```

Comment:
log the actual error. is it logger in messenger.Send ?

Reply:
It is logged at `end of loop` (with debug level):

```
// StartLoop runs the main loop
func (part *NodePart) StartLoop(request common.MessageHandler) (common.MessageHandler, error) {
	part.Messenger.SendContractRequest(request)
	response, err := part.doLoop()

	part.Logger.Debug("[NODE]: end of loop", "err", err)
	part.Messenger.ResetDialogue()
	return response, err
}
```

Because of:

```
if common.IsHookCall(message) {
	err := part.replyToHookCallRequest(message)
	if err != nil {
		return nil, err
	}

	continue
}
```